base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1120 @@
+    if (hdc == NULL)
+    {
+        DPRINT("CreateCompatibleDC(): Failed to create a valid handle!\n");
+        return FALSE;
+    }
+
+    SelectObject(hdc, BitmapMask);
+    BitBlt(lpdis->hDC, x, lpdis->rcItem.top + 1,
+           CX_BITMAP, CY_BITMAP, hdc, 0, 0, SRCAND);
+
+    SelectObject(hdc, BitmapIcon);
+    BitBlt(lpdis->hDC, x, lpdis->rcItem.top + 1,
+           CX_BITMAP, CY_BITMAP, hdc, 0, 0, SRCPAINT);
+
+    DeleteDC(hdc);
+
+    if (lpdis->itemState & ODS_FOCUS)
+    {
+        DrawFocusRect(lpdis->hDC, &lpdis->rcItem);
+    }
+
+    return TRUE;
+}
+
+void InitListViewControl(HWND hList)
+{
+    WCHAR TempList[ARR_MAX_SIZE] = { 0 };
+    WCHAR LoadedString[ARR_MAX_SIZE] = { 0 };
+
+    LVCOLUMNW lvC;
+    ZeroMemory(&lvC, sizeof(lvC));
+    lvC.mask = LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
+    lvC.cx = 158;
+    lvC.cchTextMax = 256;
+    lvC.fmt = LVCFMT_RIGHT;
+
+    SendMessageW(hList, LVM_SETEXTENDEDLISTVIEWSTYLE, 0, LVS_EX_CHECKBOXES | LVS_EX_FULLROWSELECT);
+
+    ListView_InsertColumn(hList, 0, &lvC);
+    ListView_InsertColumn(hList, 1, &lvC);
+
+    if (!CreateImageLists(hList))
+    {
+        MessageBoxW(NULL, L"CreateImageLists() failed!", L"Error", MB_OK | MB_ICONERROR);
+    }
+
+    StrFormatByteSizeW(sz.ChkDskSize, TempList, sizeof(TempList));
+    LoadStringW(GetModuleHandleW(NULL), IDS_LABEL_CHKDSK, LoadedString, _countof(LoadedString));
+    AddItem(hList, LoadedString, TempList, ICON_BLANK);
+
+    StrFormatByteSizeW(sz.RappsSize, TempList, sizeof(TempList));
+    LoadStringW(GetModuleHandleW(NULL), IDS_LABEL_RAPPS, LoadedString, _countof(LoadedString));
+    AddItem(hList, LoadedString, TempList, ICON_BLANK);
+
+    StrFormatByteSizeW(sz.RecycleBinSize, TempList, sizeof(TempList));
+    LoadStringW(GetModuleHandleW(NULL), IDS_LABEL_RECYCLE, LoadedString, _countof(LoadedString));
+    AddItem(hList, LoadedString, TempList, ICON_BIN);
+ 
+    if (bv.SysDrive)
+    {
+        StrFormatByteSizeW(sz.TempSize, TempList, sizeof(TempList));
+        LoadStringW(GetModuleHandleW(NULL), IDS_LABEL_TEMP, LoadedString, _countof(LoadedString));
+        AddItem(hList, LoadedString, TempList, ICON_BLANK);
+    }
+
+    ListView_SetCheckState(hList, 1, 1);
+    ListView_SetItemState(hList, 1, LVIS_SELECTED, LVIS_SELECTED);
+}
+
+void InitSagesetListViewControl(HWND hList)
+{
+    LVCOLUMNW lvC;
+    WCHAR LoadedString[ARR_MAX_SIZE] = { 0 };
+
+    ZeroMemory(&lvC, sizeof(lvC));
+    lvC.mask = LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
+    lvC.cx = 158;
+    lvC.cchTextMax = 256;
+    lvC.fmt = LVCFMT_RIGHT;
+    
+    SendMessageW(hList, LVM_SETEXTENDEDLISTVIEWSTYLE, 0, LVS_EX_CHECKBOXES | LVS_EX_FULLROWSELECT);
+    
+    ListView_InsertColumn(hList, 0, &lvC);
+    ListView_InsertColumn(hList, 1, &lvC);
+
+    CreateImageLists(hList);
+
+    LoadStringW(GetModuleHandleW(NULL), IDS_LABEL_CHKDSK, LoadedString, _countof(LoadedString));
+    AddItem(hList, LoadedString, L"", ICON_BLANK);
```

Comment:
if 100% of the callers of `AddItem` pass a string loaded from a resource as first string, why duplicate that loading code everywhere and not just pass the resource ID to `AddItem`, which will then load the resource string?